packages/components/src/components/date-picker/date-picker.js

```diff
@@ -40,6 +40,88 @@ Flatpickr.l10ns.en.weekdays.shorthand.forEach((day, index) => {
 
 const toArray = arrayLike => Array.prototype.slice.call(arrayLike);
 
+/**
+ * @param {number} monthNumber The month number.
+ * @param {boolean} shorthand `true` to use shorthand month.
+ * @param {Locale} locale The Flatpickr locale data.
+ * @returns {string} The month string.
+ */
+const monthToStr = (monthNumber, shorthand, locale) =>
+  locale.months[shorthand ? 'shorthand' : 'longhand'][monthNumber];
+
+/**
+ * @param {Object} config Plugin configuration.
+ * @param {boolean} [config.shorthand] `true` to use shorthand month.
+ * @param {string} config.selectorFlatpickrMonthYearContainer The CSS selector for the container of month/year selection UI.
+ * @param {string} config.selectorFlatpickrYearContainer The CSS selector for the container of year selection UI.
+ * @param {string} config.selectorFlatpickrCurrentMonth The CSS selector for the text-based month selection UI.
+ * @param {string} config.classFlatpickrCurrentMonth The CSS class for the text-based month selection UI.
+ * @returns {Plugin} A Flatpickr plugin to use text instead of `<select>` for month picker.
+ */
+const carbonFlatpickrMonthSelectPlugin = config => fp => {
+  const setupElements = () => {
+    if (!fp.monthElements) {
+      return;
+    }
+    fp.monthElements.forEach(elem => {
+      if (!elem.parentNode) return;
+      elem.parentNode.removeChild(elem);
+    });
+    fp.monthElements.splice(
```

Comment:
so we remove all of the existing `<select>` elements in the array and then insert the replacement `<span>`s here (after inserting them in the DOM before the year container) right? just curious, is there a reason we need to perform `Array.map` along with `Array.splice` or would we just be able to map through the month elements without splicing?

Reply:
Just tried to keep the array reference of `fp.monthElements`, just in case Flatpickr uses that reference in another variable. In other words, mutating the array rather than replacing with a new one.

Reply:
got it, thanks!